scripts/cmake/version.cmake

```diff
@@ -59,6 +59,27 @@ if(NOT SOF_TAG)
 	set(SOF_TAG 0)
 endif()
 
+# list tracked files from src directory
+execute_process(COMMAND git ls-files src
+		WORKING_DIRECTORY ${SOF_ROOT_SOURCE_DIRECTORY}
+		OUTPUT_FILE tracked_file_list
+	)
+# get hash of each listed files
+execute_process(COMMAND git hash-object --stdin-paths
+		WORKING_DIRECTORY ${SOF_ROOT_SOURCE_DIRECTORY}
+		INPUT_FILE tracked_file_list
+		OUTPUT_FILE tracked_file_hash_list
+	)
+# then calculate single hash of previously calculated hash list
+execute_process(COMMAND git hash-object --stdin
+		WORKING_DIRECTORY ${SOF_ROOT_SOURCE_DIRECTORY}
+		OUTPUT_STRIP_TRAILING_WHITESPACE
+		INPUT_FILE tracked_file_hash_list
+		OUTPUT_VARIABLE SOF_FW_HASH_LONG
+	)
+string(SUBSTRING ${SOF_FW_HASH_LONG} 0 8 SOF_FW_HASH)
+message(STATUS "Firmware content hash: ${SOF_FW_HASH}")
```

Comment:
for people like me who have no idea how the hash mechanism works, is the result based on a commit or would it work with temporary/non-committed changes in the files?

Reply:
>would it work with temporary/non-committed changes in the files

This one, if you wish, you can easily try it locally, just put
`git ls-files src | git hash-object --stdin-paths | git hash-object --stdin` in command line in sof root folder